src/common/encoder.rs

```diff
@@ -100,6 +100,25 @@ pub trait Encoder: Status {
     }
 }
 
+#[derive(Clone, Copy)]
+pub enum Direction {
+    Forwards,
+    Backwards,
+    StoppedForwards,
```

Comment:
do we need these two?

Reply:
maybe not? I'm using it to keep track of whether to reverse the counter once the motor is run after having been stopped